purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1770,8 +1818,8 @@ class Purchases @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE) intern
                     Store.AMAZON -> {
                         try {
                             Class.forName("com.revenuecat.purchases.amazon.AmazonBilling")
-                                .getConstructor(Context::class.java, Backend::class.java)
-                                .newInstance(application.applicationContext, backend) as BillingAbstract
+                                .getConstructor(Context::class.java, Backend::class.java, DeviceCache::class.java)
+                                .newInstance(application.applicationContext, backend, cache) as BillingAbstract
```

Comment:
not a huge deal, but I feel like we should move this to a Factory so that it alone is responsible for the reflection constructor call

Reply:
doesn't have to be for this PR though

Reply:
I will do it. I want to test this, and having it in this function makes it almost impossible